### PR TITLE
Handling ACTIVATE / DEACTIVATE blocks

### PR DESCRIPTION
**Pull request by cflynn3**

### CODE REVIEW

File: tests/Nexus/nexus_simulator/test_nexus_simulator.py [hunk 1]
```diff
@@ -1911,6 +1911,100 @@ def mock_open_wrapper(filename, mode):
...
+        TIME 01/02/2024
+          WELLS
+         NAME   STREAM DATUM 
+         welcon_1 PRODUCER 1234 
+         welcon_2 PRODUCER 5678 
+         ENDWELLS
+         
+         GASWELLS
+        NAME 		D 		DPERF 
+        gaswelcon_1		1.123e-5	INVKH
```
Comment (obi-isebor) [inline tests/Nexus/nexus_simulator/test_nexus_simulator.py:1930]: These wouldn't be new wells, but would also be welcon_1 and welcon_2

File: ResSimpy/Nexus/nexus_collect_tables.py [hunk 2]
```diff
@@ -56,13 +58,33 @@ def collect_all_tables_to_objects(nexus_file: File, table_object_map: dict[str,
...
+                                         table_start=table_start,
+                                         table_end=table_end,
+                                         is_activate_block=is_activate_block,
+                                         is_deactivate_block=is_deactivate_block,
+                                         current_date=current_date,
+                                         start_date=start_date,
+                                         date_format=date_format,
+                                         nexus_object_results=nexus_object_results)
+
+        if should_continue:
```
Comment (jonnyh404) [inline ResSimpy/Nexus/nexus_collect_tables.py:85]: Can you add a comment around which case we should continue/shouldn't continue.

File: ResSimpy/Nexus/nexus_collect_tables.py [hunk 3]
```diff
@@ -195,3 +221,62 @@ def collect_all_tables_to_objects(nexus_file: File, table_object_map: dict[str,
             table_end = -1
             token_found = None
     return nexus_object_results, nexus_constraints
+
+
+def __activate_deactivate_checks(line: str, table_start: int, table_end: int, is_activate_block: bool,
+                                 is_deactivate_block: bool, current_date: str, start_date: str | None,
+                                 date_format: DateFormat, nexus_object_results: dict[str, list[Any]]) \
+        -> tuple[bool, bool, bool]:
```
Comment (jonnyh404) [inline ResSimpy/Nexus/nexus_collect_tables.py:229]: Please add short docstring for protected method.

File: tests/Nexus/nexus_simulator/test_nexus_simulator.py
Comment (jonnyh404) [inline tests/Nexus/nexus_simulator/test_nexus_simulator.py:1]: Good test coverage.

**Review by jonnyh404 — APPROVED** [review]
Good changes. Couple of comments on docstring/comment additions.

**Pull request merged**